### PR TITLE
[hlm-707]: Delivery health service skeleton

### PR DESCRIPTION
**Pull request by shivab-egov**
Created a skeleton for delivery service.

### CODE REVIEW

File: digit-health-services/health-delivery-service/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+target/
+egov-pg-service.iml
+*.iml
+health-sync-service.iml
+src/main/resources/config/
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/.gitignore:5]: should we update the gitignore file for the new service?

File: digit-health-services/health-delivery-service/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# eGov Registration Service
+
+This module facilitates registration of beneficiaries across different campaigns and missions.
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/README.md:3]: Can we update the README.md?

File: digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/response/DeliveryResponse.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.egov.common.contract.response.ResponseInfo;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class DeliveryResponse {
+
+    @JsonProperty("responseInfo")
+    private boolean status;
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/response/DeliveryResponse.java:17]: @JsonProperty("responseInfo")
private ResponseInfo responseInfo;

Can we change

File: digit-health-services/health-delivery-service/src/main/resources/application-test.properties [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+kafka.consumer.config.auto_offset_reset=earliest
+# KAFKA PRODUCER CONFIGURATIONS
+kafka.producer.config.retries_config=0
+kafka.producer.config.batch_size_config=16384
+kafka.producer.config.linger_ms_config=1
+kafka.producer.config.buffer_memory_config=33554432
+#org.egov.detailed.tracing.enabled = false
+
+# FILE STORE SERVICE
+egov.filestore.host=http://localhost:8083
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/resources/application-test.properties:43]: We can remove unused env variables like filestore service which we may not use in this service

File: digit-health-services/health-delivery-service/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+kafka.consumer.config.auto_offset_reset=earliest
+# KAFKA PRODUCER CONFIGURATIONS
+kafka.producer.config.retries_config=0
+kafka.producer.config.batch_size_config=16384
+kafka.producer.config.linger_ms_config=1
+kafka.producer.config.buffer_memory_config=33554432
+#org.egov.detailed.tracing.enabled = false
+
+# FILE STORE SERVICE
+egov.filestore.host=http://localhost:8083
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/resources/application.properties:47]: We can remove unused env variables like filestore service which we may not use in this service

File: digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220928123130__sample_ddl.sql [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    dateOfBirth        bigint,
+    dateOfRegistration bigint,
+    md5Hash            character varying(32),
+    createdBy          character varying(64),
+    lastModifiedBy     character varying(64),
+    createdTime        bigint,
+    lastModifiedTime   bigint,
+    CONSTRAINT uk_sample_rid PRIMARY KEY (id),
+    CONSTRAINT uk_sample_hash UNIQUE (md5Hash)
+);
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220928123130__sample_ddl.sql:18]: We can remove this sql file

File: digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930151730__sync_log_create.sql [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    comment            text,
+    totalCount         integer,
+    successCount       integer,
+    errorCount         integer,
+    createdBy          character varying(64),
+    lastModifiedBy     character varying(64),
+    createdTime        bigint,
+    lastModifiedTime   bigint,
+    CONSTRAINT uk_sync_log_id PRIMARY KEY (id)
+);
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930151730__sync_log_create.sql:19]: We can remove this sql file  as the table is not fro delivery service

File: digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930153100__sync_error_details_log_create.sql [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    recordId           character varying(64),
+    recordIdType       character varying(64),
+    errorCodes         text,
+    errorMessages      text,
+    createdBy          character varying(64),
+    lastModifiedBy     character varying(64),
+    createdTime        bigint,
+    lastModifiedTime   bigint,
+    CONSTRAINT uk_sync_error_details_log_id PRIMARY KEY (id)
+);
```
Comment (jitendrac-egov) [inline digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930153100__sync_error_details_log_create.sql:15]: We can remove this sql file as the table is not for delivery service

File: digit-health-services/health-delivery-service/.gitignore (hunk 1 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/.gitignore:5]: updated

File: digit-health-services/health-delivery-service/README.md (hunk 2 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/README.md]: updated

File: digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/response/DeliveryResponse.java (hunk 3 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/response/DeliveryResponse.java]: updated.

File: digit-health-services/health-delivery-service/src/main/resources/application-test.properties (hunk 4 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/resources/application-test.properties]: updated.

File: digit-health-services/health-delivery-service/src/main/resources/application.properties (hunk 5 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/resources/application.properties]: updated.

File: digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220928123130__sample_ddl.sql (hunk 6 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220928123130__sample_ddl.sql]: updated.

**Review by jitendrac-egov — APPROVED** [review]

File: digit-health-services/health-delivery-service/egov-rn-service.yml [hunk 9]
```diff
@@ -0,0 +1,291 @@
+swagger: '2.0'
```
Comment (jayantp-egov) [inline digit-health-services/health-delivery-service/egov-rn-service.yml:1]: Remove this file

File: digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/Delivery.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class Delivery extends CampaignData{
```
Comment (jayantp-egov) [inline digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/Delivery.java:17]: We don't need this Delivery to extend CampaignData

File: digit-health-services/health-delivery-service/src/test/java/org/digit/health/delivery/web/controller/DeliveryControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+@WebMvcTest(controllers = DeliveryController.class)
+class DeliveryControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @DisplayName("should return Http status as 202 when the delivery request is valid")
```
Comment (jayantp-egov) [inline digit-health-services/health-delivery-service/src/test/java/org/digit/health/delivery/web/controller/DeliveryControllerTest.java:25]: This should be 200 OK not 202.

File: digit-health-services/health-delivery-service/src/test/resources/application-test.properties [hunk 12]
```diff
@@ -0,0 +1,40 @@
+# DATASOURCE CONFIGURATION
```
Comment (jayantp-egov) [inline digit-health-services/health-delivery-service/src/test/resources/application-test.properties:1]: Remove this file from here

File: digit-health-services/health-delivery-service/src/test/resources/application.properties [hunk 13]
```diff
@@ -0,0 +1,40 @@
+# DATASOURCE CONFIGURATION
```
Comment (jayantp-egov) [inline digit-health-services/health-delivery-service/src/test/resources/application.properties:1]: Remove this file from here

**Comment by shivab-egov** [comment]
@jayantp-egov Updated changes.

File: digit-health-services/health-delivery-service/src/test/resources/application.properties (hunk 13 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/test/resources/application.properties]: Done.

File: digit-health-services/health-delivery-service/src/test/resources/application-test.properties (hunk 12 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/test/resources/application-test.properties]: Done.

File: digit-health-services/health-delivery-service/src/test/java/org/digit/health/delivery/web/controller/DeliveryControllerTest.java (hunk 11 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/test/java/org/digit/health/delivery/web/controller/DeliveryControllerTest.java]: Done.

File: digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/Delivery.java (hunk 10 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/java/org/digit/health/delivery/web/models/Delivery.java]: Done.

File: digit-health-services/health-delivery-service/egov-rn-service.yml (hunk 9 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/egov-rn-service.yml]: Done.

File: digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930151730__sync_log_create.sql (hunk 7 above)
Comment (shivab-egov) [reply digit-health-services/health-delivery-service/src/main/resources/db/migration/main/V20220930151730__sync_log_create.sql]: done.

**Review by jayantp-egov — APPROVED** [review]

**Pull request merged**